Some (possibly opinionated) style and layout updates

enclave-alistair opened the pull request:
A few changes here while noodling around in the docs:

### Updated Menu Layout

Docs are in one page, knowledge base and tutorials are separate:

![image](https://user-images.githubusercontent.com/67738909/152346560-c1d03c1f-bfbc-4378-88d6-5d5959ab02a8.png)

Landing page is now `index.md`, so we don't always redirect to `/welcome`.

### Bullet Point Styling

Fixed bullet point styling so indentation and subsequent paragraphs align properly with the bullet, and don't use a load of calc statements to determine positioning. Change bullet colour to be primary (lighter colour felt a bit off).

Only requirement of the content is that bullets are separated by line breaks, so they become actual stylable elements, rather than just text nodes.

Before on the left, after on the right.

![image](https://user-images.githubusercontent.com/67738909/152350669-62f27099-7068-4a9e-bb0e-7be127567cd2.png)

### Styled Tags

Any text surrounded with `==` is now considered a tag (e.g. `==org.any==`), and is styled as per the portal. This helps with comprehension I think between docs and portal.

Before left, after right.

![image](https://user-images.githubusercontent.com/67738909/152350917-dc699866-214d-4318-af4c-7ad99a93b4cd.png)

enclave-marc-barry commented:
Really like this; but 

1. Suggest you revert the latest commit we just made
2. Bullet points aren't working correctly everywhere 
![image](https://user-images.githubusercontent.com/67782240/152523609-a636c537-66a1-4a60-8db9-57ebb97cddc6.png)

enclave-marc-barry commented:
Also preferred the lighter bullet colour, any reason why you darkened it?

enclave-marc-barry commented on docs/api/index.md:
```diff
@@ -31,7 +31,8 @@ are 63-character alpha-numeric strings, and are secrets that should be protected
 
-> Created tokens have access to all of the Enclave organisations that your account has access to in the portal.
+> Created tokens currently have access to all of the Enclave organisations that your account has access to in the portal. In future we will allow tokens to be restricted
+> to a specific organisation.
```
Invite readers to let us know that they want this in one of the `community-support/` channels?

enclave-marc-barry commented on docs/getting-started/troubleshooting.md:
```diff
@@ -5,22 +5,32 @@ If you find Enclave isn't working as expected, here's a simple set of troublesho
-1. Check the Enclave network adapter is online using `ipconfig /all`. Look for `Ethernet adapter Universe` in the list and check that `Media State` is not shown as `Media disconnected`. If it is, try running `enclave restart` and consider sharing local Enclave logs with [support@enclave.io](mailto:support@enclave. io) to help us investigate the problem.
+1. Check the Enclave network adapter is online using `ipconfig /all`. 
+
+    Look for `Ethernet adapter Universe` in the list and check that `Media State` is not shown as `Media disconnected`. If it is, try running `enclave restart` and consider sharing local Enclave logs with [support@enclave.io](mailto:support@enclave. io) to help us investigate the problem.
```
I will add a note to add a documentation page about how to collect logs

enclave-marc-barry replied:
Done @ https://github.com/enclave-networks/product-management/issues/209

enclave-marc-barry commented on docs/index_old.md:
```diff
@@ -0,0 +1,3 @@
+# Enclave Documentation
```
Can remove this file, we don't need it right now

enclave-marc-barry commented on mkdocs.yml:
```diff
@@ -97,21 +97,19 @@ plugins:
+      - Attach tags: getting-started/attach-tags.md
+      - Define policy: getting-started/define-policy.md
+      - Next steps: getting-started/next-steps.md
+      - Problems?:
```
I couldn't find this. Came here expecting to comment that troubleshooting had been lost from the menu.

Can we make Troubleshooting a top-level page that jumps out in the nav pls

enclave-marc-barry commented on mkdocs.yml:
```diff
@@ -59,7 +59,8 @@ markdown_extensions:
 #      linenums_style: pymdownx.inline
   - pymdownx.tabbed:
-      alternate_style: true 
+       alternate_style: true 
```
is this line over-indented now?

enclave-alistair commented:
Changed the bullet colour out of instinct I suppose, the light blue just felt a bit odd as a list-item colour.  I don't mind changing it back, but I also wonder if we should check with Dan as a tie-break?

I've fixed the bullets that weren't working, can you provided an updated screenshot were the numbers aren't centered? The developer tools seem to indicate they are.

enclave-marc-barry commented on mkdocs.yml:
```diff
@@ -76,7 +77,6 @@ plugins:
       lang: en
   - redirects:
       redirect_maps:
-        index.md: welcome.md
```
This leaves the current default document (`/welcome`) as a 404, no good. `/welcome` needs to redirect to `/` now

enclave-alistair replied:
Do we link to /welcome from somewhere? Or you just mean for legacy search links?

enclave-alistair replied:
Regardless, have added the redirect.

enclave-marc-barry commented:
Please keep the lighter bullet colour for now. Dan will come to the docs in due course and do some wider visual updates as brand matures

re; alignment - those bullet point CSS changes seem to be creating extra work for very little gain, suggest just reverting for now back to what worked and moving on

![image](https://user-images.githubusercontent.com/67782240/152526070-72ca37a6-3977-4c8b-8246-690427ce8803.png)

![image](https://user-images.githubusercontent.com/67782240/152526086-490ae647-62f1-4020-9505-dc06448dccaf.png)

enclave-alistair commented:
Accepted on colour, I'll change them back. The main reason I changed the bullet layout previously is because they weren't centrally aligned with the first line of the text, the bottom of the item shared the same baseline as the text, which made my eyes hurt.

![image](https://user-images.githubusercontent.com/67738909/152527476-2302f144-d6c5-44fd-9744-9584b6ed124e.png)

![image](https://user-images.githubusercontent.com/67738909/152527725-7f662c2a-5b75-40b4-bf69-ddd3d563f159.png)

I'll see what I can do about adjusting the vertical position of the numbers.

enclave-marc-barry commented:
> they weren't centrally aligned with the first line of the text

Damn, sorry - I assumed I'd have done a better job on that

enclave-marc-barry commented:
looking much better now, but bullet points on pages with no TOC seem a little messed up still

enclave-alistair commented:
I've tweaked all the comments above. The number central alignment is tricky due to kerning of the font mixing with the "center align" for different numbers. I think what we've got now is enough, but the 6 and 1 do 'look' slightly different in situ.  Take a look at it anyway, I'd hope it's sufficient.

To draw attention to the troubleshooting stuff, I've defined a new Support sub-menu (although I'm a little concerned it may be too high up in the list?). See what you think.

enclave-marc-barry commented:
Look out, streams crossed. Fair point on the menu ordering

enclave-alistair commented:
OK, have moved support to after setup, feels like a more natural place.

enclave-marc-barry commented:
Nice, menu ordering looks better - although the bullet points still look out of whack at the bottom of the `/` page (they look fine on all of the other pages without TOC however)?

enclave-alistair commented:
OK, updated CSS for bullets where no `<p>` tag is added. Believe that should fix all remaining bullet points. Have I mentioned I hate CSS?

enclave-marc-barry reviewed (approved):
Looks great (I added some minor additional commits btw)
